mesa/space.py

```diff
@@ -419,17 +419,61 @@ def place_agent(self, agent: Agent, pos: Coordinate) -> None:
     def remove_agent(self, agent: Agent) -> None:
         ...
 
-    def move_agent(self, agent: Agent, pos: Coordinate) -> None:
-        """Move an agent from its current position to a new position.
+    def move_agent(
+        self,
+        agent: Agent,
+        pos: Coordinate | list[Coordinate],
+        selection: str = "random",
```

Comment:
One thought on UX here. 

I am thinking it should be explicit that the user has to make a choice (i.e. `random` or `closest` when they pass in a list of coordinates. I think this will ensure they understand at least part if their movement behavior. 

So maybe the default is `move_agent(agent, user_specified_position, selection ="user_specified") and if the user doesn't specify (i.e. passes in a list of coordinates) then it tells they need to pass in a coordinate or choose `random` or `closest`

 Thoughts?

Reply:
In my view, if pos is a single location, selection becomes irrelevant. So, the default value is `selection=None`. If pos is a collection and `selection=None`,  you raise an error (probably a `ValueError`).

The interesting corner case is a list with 1 position and how to handle this. In my view, you handle this as a list, so `selection` should not be `None`.

Reply:
Personally, I think random is a good default. You input a list, it chooses one without bias or favor.

I was in doubt of even including closest, but I thought it was an interesting addition for some edge-cases.

Reply:
My vote is to go with `selection=None`  and throw a `ValueError` if they pass a list without changing selection. This ensures the users choices are explicit and understood (as much as possible). 

@EwoutH My assumption is this is not a heavy lift, would you be able to make that update? If my assumption is wrong please let me know or if you feel passionately about not doing this.

Reply:
Technically, it's an easy change.

I just don't like the boiler plate we're adding, while in my opinion an unbiased (thus random) selection is a very obvious default. If you specify some criteria (like lowest distance), you specify it, if not, it selects without a criteria randomly.

Most of the use cases I can think of, I will already have narrowed down the possible locations I can go, for example to a neighbourhood or all values above a threshold in the property layer. So I want to just pick one of the, without favor. That makes random in my opinion the obvious default choice.

Reply:
But that is only true if the position is a list. If you pass a single position, the keyword argument random is meaningless. Moreover, I'll be happy to debate what is obvious in this case. For you, it might be random, but for others working in other domains, closest might be the obvious. So, the pythonic response, in my view, is to refuse to guess.

Reply:
In my view, you have two categories of selection:
- Unfavored/undirected: Random
- Favored/directed: Closest, farthest, highest, quietest, etc.

Note that unfavored only has one option: completely random. Any other selection you need some selection criteria. I would argue that unfavored selection a good default, and if you want to introduce favor/direction you have to specify the criteria what you are going to favor.

Maybe we should first discuss if this PR needs to modify `move_agent` or should introduce a new method (see my comment below).

Reply:
Closest and farthest just require a distance metric. So, if you want to be complete, you probably would need to pass some additional thing(s) in case of non-random selection. 

I agree that the additional method warrants its own discussion.

Reply:
I agree with @Ewouth's viewpoint. When given a list of potential locations, selecting one at random should be the default approach. Why? If the intention was to always pick the first item, there wouldn't be a need for a list. And if specific criteria for selection exist, they must be clearly defined. In scenarios lacking such criteria, a random choice is the most logical. This logic applies even to a list with a single item; selecting randomly from one option inevitably results in that option being chosen.

Regarding an empty list, the concept is straightforward: if you randomly select from an empty list, you end up with no new position, meaning you remain in your current location. It's conceivable to have a system that generates a list of possible positions, which could be empty in some cases.

Reply:
I have found random to be the easy explain away with ABMs but in real life I don't see it, it is typically not random, typically it is "the path of least resistance." (not to mention the endless debates of computer scientists on the "randomness" of different random generators) So I am firmly with @quaquel on this and will emphasize his quote  "So, the pythonic response, in my view, is to refuse to guess."

However we are tied 2 to 2. 

If everyone is ok with it I can ping @jackiekazil Jackie and have her tiebreak.

Reply:
> If everyone is ok with it I can ping Jackie and have her tiebreak.

Yes, but I would like the discussion if moving from a list (and possibly non-moving on an empty list) should be a separate method or not. Because it will influence my standpoint and it might influence others.

If we go with allowing an empty list input and thus allowing non-movements, I think it should be a separate method.

Reply:
To make sure I understand, @EwoutH your proposal is just to make two separate methods. 
1 - move agent where parameter is the position to move to
2 - move agent where a list is passed in as the argument and then random is the default selection

If I am understanding, I think that is great, I would totally support that. That solves the entire discussion IMHO.

Reply:
Yes, that’s indeed what I’m proposing.

I will adjust this PR to reflect that.

Reply:
It seems like this is resolved, and if I understand it correctly I am in agreement. Let me know if I am misunderstanding.